emanate/config.py

```diff
@@ -0,0 +1,92 @@
+import collections.abc
+import json
+from pathlib import Path
+
+
+DEFAULT_SETTINGS = {
+    'ignore': frozenset((
+        "*~",
+        ".*~",
+        ".*.sw?",
+        "*/emanate.json",
+        "*.emanate",
+        ".*.emanate",
+        "*/.git",
+        "*/.git/*",
+        "*/.gitignore",
+        "*/.gitmodules",
+        "*/__pycache__",
+        "*/__pycache__/*",
+    )),
+
+    'clean': False,
+    'confirm': True,
+    'destination': Path.home(),
+}
+
+
+# Boilerplate stolen from Cumin:
+#   https://github.com/astronouth7303/cumin/blob/master/cumin/config.py
+class Config(collections.abc.MutableMapping):
+    """
+    Configuration that just initializes itself from default values.
+    """
+
+    def __init__(self, **kwargs):
+        if not kwargs:
+            self._data = {}
+        else:
+            self._data = kwargs
+
+    def __getitem__(self, key):
+        return self._data.get(key, DEFAULT_SETTINGS[key])
```

Comment:
This should be `DEFAULT_SETTINGS.get(key)`, or you can only have settings you have defaults for.

Reply:
Right, thanks (that's the case currently, but not necessarily in general)